aten/src/ATen/native/cuda/ScatterGatherKernel.cu

```diff
@@ -327,10 +293,6 @@ struct cuda_scatter_fill_base_kernel {
     const std::string& method_name,
     const func_t& f
   ) {
-    // no-op if index is empty
-    if (index.numel() == 0) {
-      return;
-    }
     at::assert_no_internal_overlap(self);
 
     dim = maybe_wrap_dim(dim, self.dim());
```

Comment:
After adding the `maybe_wrap_dim` call to them IMPL functions, you won't need these anymore.